rhessys/tec/input_new_strata.c

```diff
@@ -86,174 +92,124 @@ void input_new_strata(
- 		fscanf(world_file,"%lf",&(ltmp));
-		read_record(world_file, record);
-		if (fabs(ltmp - NULLVAL) >= ONE) canopy_strata[0].ns.leafn_transfer = ltmp;
- 		fscanf(world_file,"%lf",&(ltmp));
-		read_record(world_file, record);
-		if (fabs(ltmp - NULLVAL) >= ONE) canopy_strata[0].ns.live_stemn = ltmp;
- 		fscanf(world_file,"%lf",&(ltmp));
-		read_record(world_file, record);
-		if (fabs(ltmp - NULLVAL) >= ONE) canopy_strata[0].ns.livestemn_store = ltmp;
- 		fscanf(world_file,"%lf",&(ltmp));
-		read_record(world_file, record);
-		if (fabs(ltmp - NULLVAL) >= ONE) canopy_strata[0].ns.livestemn_transfer = ltmp;
- 		fscanf(world_file,"%lf",&(ltmp));
-		read_record(world_file, record);
-		if (fabs(ltmp - NULLVAL) >= ONE) canopy_strata[0].ns.dead_stemn = ltmp;
- 		fscanf(world_file,"%lf",&(ltmp));
-		read_record(world_file, record);
-		if (fabs(ltmp - NULLVAL) >= ONE) canopy_strata[0].ns.deadstemn_store = ltmp;
- 		fscanf(world_file,"%lf",&(ltmp));
-		read_record(world_file, record);
-		if (fabs(ltmp - NULLVAL) >= ONE) canopy_strata[0].ns.deadstemn_transfer = ltmp;
- 		fscanf(world_file,"%lf",&(ltmp));
-		read_record(world_file, record);
-		if (fabs(ltmp - NULLVAL) >= ONE) canopy_strata[0].ns.live_crootn = ltmp;
- 		fscanf(world_file,"%lf",&(ltmp));
-		read_record(world_file, record);
-		if (fabs(ltmp - NULLVAL) >= ONE) canopy_strata[0].ns.livecrootn_store = ltmp;
- 		fscanf(world_file,"%lf",&(ltmp));
-		read_record(world_file, record);
-		if (fabs(ltmp - NULLVAL) >= ONE) canopy_strata[0].ns.livecrootn_transfer = ltmp;
- 		fscanf(world_file,"%lf",&(ltmp));
-		read_record(world_file, record);
-		if (fabs(ltmp - NULLVAL) >= ONE) canopy_strata[0].ns.dead_crootn = ltmp;
- 		fscanf(world_file,"%lf",&(ltmp));
-		read_record(world_file, record);
-		if (fabs(ltmp - NULLVAL) >= ONE) canopy_strata[0].ns.deadcrootn_store = ltmp;
- 		fscanf(world_file,"%lf",&(ltmp));
-		read_record(world_file, record);
-		if (fabs(ltmp - NULLVAL) >= ONE) canopy_strata[0].ns.deadcrootn_transfer = ltmp;
- 		fscanf(world_file,"%lf",&(ltmp));
-		read_record(world_file, record);
-		if (fabs(ltmp - NULLVAL) >= ONE) canopy_strata[0].ns.frootn = ltmp;
- 		fscanf(world_file,"%lf",&(ltmp));
-		read_record(world_file, record);
-		if (fabs(ltmp - NULLVAL) >= ONE) canopy_strata[0].ns.frootn_store = ltmp;
- 		fscanf(world_file,"%lf",&(ltmp));
-		read_record(world_file, record);
-		if (fabs(ltmp - NULLVAL) >= ONE) canopy_strata[0].ns.frootn_transfer = ltmp;
- 		fscanf(world_file,"%lf",&(ltmp));
-		read_record(world_file, record);
-		if (fabs(ltmp - NULLVAL) >= ONE) canopy_strata[0].ns.cwdn = ltmp;
- 		fscanf(world_file,"%lf",&(ltmp));
-		read_record(world_file, record);
-		if (fabs(ltmp - NULLVAL) >= ONE) canopy_strata[0].ns.retransn = ltmp;
-		
+	default_object_ID = getIntWorldfile(&paramCnt,&paramPtr,"veg_parm_ID","%d",0,1);
+	ltmp = getDoubleWorldfile(&paramCnt,&paramPtr,"cover_fraction","%lf",canopy_strata[0].cover_fraction,1);	
```

Comment:
Brian,
      Thanks for reminding me.

      The variables using to store the value of 'soil_parm_ID',
'landuse_parm_ID', etc are temporary variables in init/construct_patch.c
and init/construct_canopy_stratum.c .

      I will add it to the patch[] data structure and let it get the value
from init/construct_**, and call it in tec/input_new**.c.

      How do you like it?

Best,
Xiaoli

On Wed, Apr 6, 2016 at 6:54 AM, Brian Miles <notifications@github.com>
wrote:

> In rhessys/tec/input_new_strata.c
> <https://github.com/RHESSys/RHESSys/pull/82#discussion_r58708610>:
>
> > -		if (fabs(ltmp - NULLVAL) >= ONE) canopy_strata[0].ns.frootn = ltmp;
> > - 		fscanf(world_file,"%lf",&(ltmp));
> > -		read_record(world_file, record);
> > -		if (fabs(ltmp - NULLVAL) >= ONE) canopy_strata[0].ns.frootn_store = ltmp;
> > - 		fscanf(world_file,"%lf",&(ltmp));
> > -		read_record(world_file, record);
> > -		if (fabs(ltmp - NULLVAL) >= ONE) canopy_strata[0].ns.frootn_transfer = ltmp;
> > - 		fscanf(world_file,"%lf",&(ltmp));
> > -		read_record(world_file, record);
> > -		if (fabs(ltmp - NULLVAL) >= ONE) canopy_strata[0].ns.cwdn = ltmp;
> > - 		fscanf(world_file,"%lf",&(ltmp));
> > -		read_record(world_file, record);
> > -		if (fabs(ltmp - NULLVAL) >= ONE) canopy_strata[0].ns.retransn = ltmp;
> > -		
> > +	default_object_ID = getIntWorldfile(&paramCnt,&paramPtr,"veg_parm_ID","%d",0,1);
> > +	ltmp = getDoubleWorldfile(&paramCnt,&paramPtr,"cover_fraction","%lf",canopy_strata[0].cover_fraction,1);	
>
> @xiaoli-chen <https://github.com/xiaoli-chen> For input_new_X_*(), the
> default value of each variable should be the current value of the variable.
> This way, redefine worldfiles only need to contain the variables to be
> redefined.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/RHESSys/RHESSys/pull/82/files/64ebd7925e091cf4e82fd6471189e9b4da2aa1fc#r58708610>
>



-- 
Xiaoli Chen

Bren School of Environmental Science and Management
University of California, Santa Barbara
xiaoli_chen@umail.ucsb.edu